courses/management/commands/upgrade_eligible_users.py

```diff
@@ -62,6 +62,7 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
                     user=user, run=course_run
                 ).first()
                 if enrollment is None:
+                    users_no_enrollment.append((row[0], course_run.course_id))
```

Comment:
This should be the `courseware_id` for the course run - if it's numeric the `generate_legacy_enrollment_codes` command will look it up in the MicroMasters import CourseId table, but it'll look it up based on the MicroMasters ID so coupon generation later will fail. Using the readable ID for the run will work, though.